### DISCUSSION TITLE
Layer image won't be inserted

### DISCUSSION THREAD

**Discussion by ZJulienZ** [sn4k3/UVtools#749] [Q&A] [answered]
Hello there,
I have been enjoying using UVtools for the last couple of weeks but I am now facing a new issue.
As I am trying to find the right printing parameters for my purpose, the process I have been using without any issue is:
1) generate a layer image (bmp) with various patterns (8bit bitmap)
2) import that image in an existing file (action->import Layer)
3) twick the printing parameters
4) print
I've had no problem whatsoever so far but the last image I created (file attached) will not be inserted by "import layer" and I have no error message.
I've tried opening it and saving it with several softwares even with UVTools but even the png image generated by UVtools won't be inserted....
is there a way I can get the logs of what the program is doing to find what is the issue with this file?
[ScanPost198-110umSpace330-22um_Grids110_22-110_Labelled_Flipped.bmp.zip](https://github.com/sn4k3/UVtools/files/12485340/ScanPost198-110umSpace330-22um_Grids110_22-110_Labelled_Flipped.bmp.zip)
thanks a lot!!

**Comment by sn4k3** [comment]
I'm able to import your image into a pm5s file.

The import tool will ignore all images that he can't import for any reason (eg. lack of space).
I'm guessing you using the "Replace" method. Your image is 7860 x 4320, check if you have sufficient space on X/Y to import that. 
If you still have the problem, attach the sliced file so I can test on it.

Note: If you wish, use PNG instead to save many MB's

**Reply by ZJulienZ** [reply, marked as answer]
Hello,
Just to let you know how I worked around that issue:
I tried png (which indeed makes much smaller files to import) but without success so I guess it was not  a "RAM emergency".
I am using the Insert method (but the stake and replace didn't work either).
The printer is a Phrozen Sonic Mini 8K so the image I insert is the size of the LCD and it was working so far without any problem so it is not a X/Y "space issue".
The problem is definetly on the bmp file side: I was able to insert  the one I generated with my pattern generator (matrix generated and saved as bmp by labview), or the one saved from Photoshop after I added some labels and flipped the image horizontally. But if I make other manipulation of the image in Photoshop (in that case cut/paste portions of image then flatten the image then save as bmp) then the image will not be inserted no matter what, even if I open it with UVTools and save it from UVTools before import.
So I guess, somewhere along the way Photoshop modified some header/metadata of the file that made it not enough corrupted to prevent UVTools from reading it, but enough so that it can't import it. I would guess that there are two different routine for header handling for open/save the bmp and for import and the devil is in those differences.
But still thanks a thousand for the great work with providing to the community with the superbe tools that is UVTools, it really saved me I my current project.

**Reply by ZJulienZ** [reply]
I have the answer why !!
For some reason, when looking for the LCD specs of my printer, because they are not clearly given by the company, I first came upon a value of 7680x4320 pixels which is NOT what the particular Phrozen Sonic Mini 8K screen is: a 7500x3240 (as in machine preset values you have in UVTools). So the bmp images I was generating were out of size with the ctb file I was trying to insert them in.
If the pattern was not too large, apparently UVTools managed to trim them in a layer but after adding some labels and what-not in Photoshop then non-zero pixels would be in the cropped region and that would cause it not to be imported.
Now that I have the right size in my pattern generator I can import the bmp image seamlessly.

**Reply by sn4k3** [reply]
When you open any file on UVtools it tells you the real resolution and screen sizes on bottom right, always go with that.
